.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [ '1.22', '1.23.x' ]
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Build
+        run: go build -v ./...
```

Comment:
I didn't see this in our instructions, possibly it's not necessary? or is the idea to check that our code builds in addition to the tests passing? not sure how the tests could pass if the build fails

Reply:
Indeed, it's not necessary, the next step would be fine by itself.
The idea of triggering a build of the code-base, is to download and build the dependencies.
The Go toolkit is good at caching, most of this work will be reused in the following step.

The main benefit is to be able to observe how long the download/build step took, independently from the test step.

It's useful for code-base with a lot of dependencies, but it may not be the case for this project.
We could remove it, but honestly, the overhead is negligible.